phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java

```diff
@@ -461,9 +463,9 @@ private static String toString(List<byte[]> gps) {
         for(Pair<byte[], byte[]> where : context.getWhereConditionColumns()) {
           whereConditions.add(where.getFirst());
         }
-        List<byte[]> gps = getGuidePosts(whereConditions);
+        GuidePostsInfo gps = getGuidePosts(whereConditions);
         if (logger.isDebugEnabled()) {
-            logger.debug("Guideposts: " + toString(gps));
+            logger.debug("Guideposts: " + gps);
         }
```

Comment:
Minor: I'd get rid of this logger if statement. The output could be pretty huge.